app/modules/enduser_setup.c

```diff
@@ -626,6 +626,52 @@ static void do_station_cfg (task_param_t param, uint8_t prio)
 }
 
 
+/**
+ *
+ *
+ *
+ *
+ */
+static int enduser_setup_write_file_with_extra_configuration_data(
+char * configuration_string, int configuration_length
+)
+{
+  int p_file = 0;
+  lua_State *L = NULL;
+	
+  ENDUSER_SETUP_DEBUG("enduser: opening enduser_custom_parameters.json for write");
+
+  // setup the file output
+  p_file = vfs_open("enduser_custom_parameters.json", "w");
+  if (p_file == 0)
+  {
+    ENDUSER_SETUP_DEBUG("Can't write to file!");
+    return 1;
+  }
+  
+  /* Not certain what this does */
+  L = lua_getstate();
+  
+  if (L != NULL) 
+  {
+    lua_pushlstring(L, configuration_string, configuration_length);
+  }
+  
+  vfs_write(p_file, configuration_string, configuration_length);
+  vfs_close(p_file);
+/*  
+  if (L != NULL) 
+  {
+    ENDUSER_SETUP_DEBUG("enduser: calling lua for validation");
+
+    lua_call(L, 1, 1);
```

Comment:
The idea of a validation callback is good, but you'll need to pay attention to the return value in such case, and also the file shouldn't be written if the validation fails. If the callback is just for notification the current location is good. You'll also need to actually push the callback function onto the stack before the argument in order to be able to `lua_call()` it. There are plenty of examples hanging around in the modules, have a look for `lua_call()`, `luaL_ref()`, `luaL_unref()` and `lua_rawgeti()`.